tests/src/test/scala/cats/tests/FunctionKSuite.scala

```diff
@@ -92,12 +92,27 @@ class FunctionKSuite extends CatsSuite {
   }
 
   test("lift compound unary") {
-    val fNelFromList = FunctionK.lift[List, λ[α => Option[NonEmptyList[α]]]](NonEmptyList.fromList _)
+    val fNelFromList = FunctionK.lift[List, OptionOfNel](NonEmptyList.fromList)
     forAll { (a: List[String]) =>
       fNelFromList(a) should ===(NonEmptyList.fromList(a))
     }
   }
 
+  test("lift eta-expanded function") {
+    val fSomeNel = FunctionK.lift[NonEmptyList, OptionOfNel](Applicative[Option].pure)
+    forAll { (a: NonEmptyList[Int]) =>
+      fSomeNel(a) should ===(Some(a))
+    }
+  }
+
+  test("lift a function directly") {
+    def headOption[A](list: List[A]): Option[A] = list.headOption
+    val fHeadOption = FunctionK.liftFunction[List, Option](headOption)
```

Comment:
A question for my understanding, could this be
 ```suggestion
     val fHeadOption = FunctionK.lift[List, Option](headOption)
```
Isn't it forwarding to `liftFunction` anyway ? Or it's testing the public `liftFunction` explicitly on purpose ?
Apologies if this doesn't make sense :).

Reply:
It's testing `liftFunction` explicitly. Do you think we should also try to add a dotty version? It has native polymorphic functions, but we don't provide a way to convert them to `FunctionK` and I think type inference is still not great.

Reply:
It makes sense, thanks for clarifying. About the dotty version, I guess it's related to https://github.com/typelevel/cats/issues/2553 that's still an open issue for what I understand and specifically there is work going on as note here https://github.com/typelevel/cats/issues/2553#issuecomment-561076000. We could sync a dotty specific version with that effort.
So what do you thing (you and others) about letting this PR move on as it improves the current situation described here https://github.com/typelevel/cats/issues/3400 and might make some user's life easier ? My approval goes in that direction. Of course I happy to do otherwise if the general preference is different.

Reply:
Yes I agree this can be merged and I can try to add a dotty version in another PR 👍

Reply:
Brilliant, thank 🙇